.github/workflows/compilation.yml

```diff
@@ -44,18 +44,18 @@ jobs:
       fail-fast: false
       matrix:
         config:
-        - { name: "gcc 14",   cc: gcc-14,   tidy: "On"  }
```

Comment:
GCC has tidy turned off now?

Reply:
Well, I guess this is about clang-tidy, so it would be a mistake to have it on for gcc.

Reply:
I've moved the clang-tidy run from gcc unittests to clang unittests

clang-tidy can only consume the precompiled header files generated by a matching version of clang
Enabling Tidy in a gcc build will disable PCH and emit a warning (I added a check in cmakelists.txt)
This is also why I bumped clang from 16 to 18, to match our clang-tidy version.